### PR TITLE
Add test for listener.tlsIdentity property and refactor listener clean up in tests

### PR DESCRIPTION
**Pull request by pasin**
* Add test for URLEndpointListener.tlsIdentity property.
* Refactor how to clean up or stop the listener in tests by using stopListen or stopListener:.

### CODE REVIEW

File: Objective-C/Tests/URLEndpointListenerTest.m [hunk 1]
```diff
@@ -589,8 +660,7 @@ - (void) testUnavailableNetworkInterface {
     [self ignoreException:^{
         [self listen: config errorCode: CBLErrorUnknownHost errorDomain: CBLErrorDomain];
     }];
-
-    [_listener stop];
+    [self stopListen];
```
Comment (jayahariv) [inline Objective-C/Tests/URLEndpointListenerTest.m:663]:
```suggestion
```

File: Objective-C/Tests/URLEndpointListenerTest.m [hunk 2]
```diff
@@ -735,6 +798,8 @@ - (void) testReplicatorAndListenerOnSameDatabase {
     repl2 = nil;
     Assert([db2 close: &err], @"Failed to close db2 %@", err);
     db2 = nil;
+    
+    [self stopListen];
```
Comment (jayahariv) [inline Objective-C/Tests/URLEndpointListenerTest.m:802]: teardown calls same?
```suggestion
```

File: Objective-C/Tests/URLEndpointListenerTest.m [hunk 3]
```diff
@@ -768,7 +836,9 @@ - (void) testCloseWithActiveListener {
     
     AssertEqual(_listener.port, 0);
     AssertEqual(_listener.urls.count, 0);
+    
+    // cleanup
+    [self stopListen];
```
Comment (jayahariv) [inline Objective-C/Tests/URLEndpointListenerTest.m:841]: ?
```suggestion
```

File: Objective-C/Tests/URLEndpointListenerTest.m (hunk 2 above)
Comment (pasin) [reply Objective-C/Tests/URLEndpointListenerTest.m:802]: Yes, just in case a test forgets to stop listen.

**Pull request merged**